lua/telescope/builtin/__lsp.lua

```diff
@@ -211,6 +150,22 @@ local function list_or_jump(action, title, opts)
   end)
 end
 
+lsp.references = function(opts)
+  local filepath = vim.api.nvim_buf_get_name(opts.bufnr)
+  local lnum = vim.api.nvim_win_get_cursor(opts.winnr)[1]
+  local include_current_line = vim.F.if_nil(opts.include_current_line, false)
+
+  -- If we shouldn't include the current line, add a function to filter it out
+  -- from the LSP results.
+  if not include_current_line then
+    opts.results_filter = function(v)
+      return not (v.filename == filepath and v.lnum == lnum)
+    end
+  end
+
+  return list_or_jump("textDocument/references", "LSP References", opts)
```

Comment:
```lua
params.context = { includeDeclaration = vim.F.if_nil(opts.include_declaration, true) }
```

we can either pass in params as opts.

Reply:
Not sure what this is referring to, sorry!

Reply:
I think this comment is referring to the need to support the `include_declaration` option for `lsp_references`.

See the following code:
https://github.com/nvim-telescope/telescope.nvim/blob/5fff2a138b04be7e23959c1da5e535bdac7e1518/lua/telescope/builtin/__lsp.lua#L17-L19

Reply:
@gbroques thanks for the clarification! I added some logic in `list_or_jump` to handle this case.